js/components/directionControl.js

```diff
@@ -0,0 +1,45 @@
+var React = require('react');
+var classnames = require('classnames');
+
+var DirectionControl = React.createClass({
+  getInitialState: function() {
+    return {
+      isTouched: false,
+    };
+  },
+
+  componentWillUnmount: function() {
+    this.timer && clearTimeout(this.timer);
+  },
+  
+  handleEvent: function (ev) {
+    var rotate = ev.type == 'mousedown' || ev.type == 'touchstart';
+    this.props.handleDirection(rotate, this.props.dir);
+    
+    this.setState({
+      isTouched: true
+    }, function(){
+      var _this = this;
+      this.timer = setTimeout(function(){
```

Comment:
Do we need to clear the current timer before starting a new one?

Reply:
\+